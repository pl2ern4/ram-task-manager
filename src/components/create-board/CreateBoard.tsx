import {Button} from '@mui/material';
import { SetStateAction, useState } from 'react';
import {CreateBoardForm, CustomInput} from './styles';
const VALID_BOARD_STRING = /^[A-Za-z0-9 _]*[A-Za-z0-9][A-Za-z0-9 _]*$/;

type Props = {
    setBoardName:(value:string)=>void
}
const CreateBoard = ({setBoardName}: Props) => {
    const [value, setValue] = useState('');
    const handleChange = (e: { target: { value: SetStateAction<string>; }; }) =>{
        setValue(e.target.value);
    }
    const handleSubmit = () =>{
        if(VALID_BOARD_STRING.test(value)){
            setBoardName(value)
        }
    }
    return (
        <CreateBoardForm fullWidth focused className='FormControl'>
            <CustomInput name="board-input" inputProps={{maxLength:20}} value={value} onChange={handleChange} required autoFocus/>
            <Button variant="contained" onClick={handleSubmit}>Create Board</Button>
        </CreateBoardForm>
    )
}

export default CreateBoard;
