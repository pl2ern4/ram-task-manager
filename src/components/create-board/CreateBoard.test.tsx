import '@testing-library/jest-dom'
import { render } from '@testing-library/react'
import CreateBoard from './CreateBoard';

describe('CreateBoard', () => {
    const setBoardNameFn = jest.fn();
    it('should call `setBoardName`, with Valid input', async () => {
        const {container} = render(<CreateBoard setBoardName={setBoardNameFn}/>)
        expect(container).toMatchSnapshot();
    });
})