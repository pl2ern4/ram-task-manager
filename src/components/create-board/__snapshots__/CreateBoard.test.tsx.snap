// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateBoard should call \`setBoardName\`, with Valid input 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiFormControl-fullWidth FormControl css-1s8w3h6-MuiFormControl-root"
  >
    <div
      class="MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-colorPrimary Mui-focused MuiInputBase-formControl css-1j7cbzm-MuiInputBase-root-MuiInput-root"
      required=""
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiInput-input css-1x51dt5-MuiInputBase-input-MuiInput-input"
        maxlength="20"
        name="board-input"
        required=""
        type="text"
        value=""
      />
    </div>
    <button
      class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-colorPrimary css-sghohy-MuiButtonBase-root-MuiButton-root"
      tabindex="0"
      type="button"
    >
      Create Board
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</div>
`;
